pkg/controllers/multiclusterservice/mcs_controller.go

```diff
@@ -119,6 +119,46 @@ func (c *MCSController) Reconcile(ctx context.Context, req controllerruntime.Req
 	return controllerruntime.Result{}, nil
 }
 
+func (c *MCSController) checkClusterConfiguration(mcs *networkingv1alpha1.MultiClusterService,
+	providerClusters, noneExistProviderClusters, consumerClusters, noneExistConsumerClusters sets.Set[string]) error {
+	if len(noneExistProviderClusters) != 0 || len(noneExistConsumerClusters) != 0 {
+		msgProvider := strings.Join(noneExistProviderClusters.UnsortedList(), ",")
+		msgConsumer := strings.Join(noneExistConsumerClusters.UnsortedList(), ",")
+		c.EventRecorder.Eventf(mcs, corev1.EventTypeNormal, events.EventReasonConfigurationRedundant,
+			fmt.Sprintf("ProviderClusters(%s)/ConsumerClusters(%s) dont's exist, Karmada will ignore them", msgProvider, msgConsumer))
+	}
+
+	if !helper.MultiClusterServiceCrossClusterEnabled(mcs) {
+		return nil
+	}
```

Comment:
For MultiClusterService that not enabled `CrossCluster` already has been ignored at the `mcsPredicateFunc`, why check again here?

Reply:
No, we don't ignore it. The condition is:
```
if !helper.MultiClusterServiceCrossClusterEnabled(mcsOld) && !helper.MultiClusterServiceCrossClusterEnabled(mcsNew) {
....
```